ocs_ci/ocs/longevity_stage2.py

```diff
@@ -0,0 +1,563 @@
+
+    Logs:
+        PVC Creation Time of all the PVCs.
+
+    """
+    msg_prefix = f"Interface: {interface}, PVC size: {pvc_size}."
+    accepted_creation_time = 1
+    pvc_no = 0
+    for pvc_obj in pvc_objs:
+        creation_time = performance_lib.measure_pvc_creation_time(
+            interface, pvc_obj.name, start_time
+        )
+
+        logger.info(
+            f"{msg_prefix} PVC number {pvc_no + 1} of interface {interface} was created in {creation_time} seconds."
+        )
+        if creation_time > accepted_creation_time:
+            raise ex.PerformanceException(
+                f"{msg_prefix} PVC creation time is {creation_time} and is greater than "
+                f"{accepted_creation_time} seconds."
+            )
+        pvc_no += 1
+
+
+def measure_pvc_deletion_time(interface, num_of_pvcs, pvc_size, pvc_objs, pv_objs):
+    """
+    Measures and Logs PVC Deletion Time of all PVCs.
+
+    Args:
+        interface (str) : an interface (RBD or CephFS) to run on.
+        num_of_pvcs (int) : Number of PVCs we have created.
+        pvc_size (int) : Size of each PVC in GB.
+        pvc_objs (list) : List of PVC objects for which we have to measure the time.
+        pv_objs (list) : List of PV names created for the respective PVCs.
+
+    Raises:
+        PerformanceException : Raises an exception if PVC deletion time is greater than the accepted time.
+
+    Logs:
+        PVC Creation Time of all the PVCs.
+
+    """
+    msg_prefix = f"Interface: {interface}, PVC size: {pvc_size}."
+    accepted_deletion_time = 1
+    if interface == constants.CEPHFILESYSTEM:
+        accepted_deletion_time = 2
+    elif interface == constants.CEPHBLOCKPOOL:
+        accepted_deletion_time = 1
+
+    for pvc_no in range(num_of_pvcs):
+        pvc = pvc_objs[pvc_no]
+        pv = pv_objs[pvc_no]
+        pvc.ocp.wait_for_delete(resource_name=pvc.name)
+        helpers.validate_pv_delete(pv)
+        deletion_time = helpers.measure_pvc_deletion_time(interface, pv)
+        logger.info(
+            f"{msg_prefix} PVC number {pvc_no + 1} of interface {interface} was deleted in {deletion_time} seconds."
+        )
+        if deletion_time > accepted_deletion_time:
+            raise ex.PerformanceException(
+                f"{msg_prefix} PVC deletion time is {deletion_time} and is greater than "
+                f"{accepted_deletion_time} seconds."
+            )
+
+
+def _multi_pvc_pod_lifecycle_factory(multi_pvc_factory, pod_factory, teardown_factory):
+    """
+    Creates a factory that is used to:
+    1. Create/Delete PVCs of type:
+        a. CephFileSystem - RWO
+        b. CephFileSystem - RWX
+        c. CephBlockPool - RWO
+        d. CephBlockPool - RWO - Block
+        e. CephBlockPool - RWX - Block
+    2. Create/Delete PODs.
+    2. Measure the PVC creation/deletion time and POD attach time.
+
+    """
+
+    def factory(num_of_pvcs=100, pvc_size=2, bulk=False, namespace="stage-2"):
```

Comment:
IMO, it is better to set the time measurement as an optional arg. Time measurement is not a must for all tests

Reply:
Done. I've set a measure flag, which is set to True if we want to measure the performance metrics, False otherwise.